src/components/ThreeDS2/components/utils.ts

```diff
@@ -1,11 +1,26 @@
 import { ERROR_MESSAGES, ERRORS, CHALLENGE_WINDOW_SIZES } from '../config';
 import { getOrigin } from '../../../utils/getOrigin';
 import base64 from '../../../utils/base64';
+import { ChallengeData, ChallengeToken, FingerPrintData, ResultObject } from '../types';
 
-export const decodeAndParseToken = token => {
+interface ResolveData {
+    data: {
+        details: {
+            [key: string]: string;
+        };
+        paymentData: string;
+    };
+}
+
+interface ErrorCode {
+    errorCode?: string;
+    message?: string;
+}
+
+export const decodeAndParseToken = (token: string): ChallengeToken => {
     const decodedToken = base64.decode(token);
     try {
-        return JSON.parse(decodedToken);
+        if (decodedToken) return JSON.parse(decodedToken);
```

Comment:
Is the `if` needed? That's why this is in a `try...catch`

Reply:
Yes, but just because `base64.decode` can return either a string or `false`, while JSON.parse expects a string, so TS was complaining about that.

Reply:
Fair enough!